projects/storefrontlib/shared/components/item-counter/item-counter.component.ts

```diff
@@ -78,8 +80,15 @@ export class ItemCounterComponent implements OnInit, OnDestroy {
    */
   private sub: Subscription;
 
+  constructor(
```

Comment:
Although there is still a lot of legacy code left behind with constructor injection, in general, we have stopped using the constructor to inject dependencies. This is because it is easier to run our migration scripts. Instead, we use injector syntax.

```
protected featureConfigService = inject(FeatureConfigService, {                                                                    
    optional: true,
});
```

**Why is this easier for our migration scripts?**

Because our migration scripts read try to read changes in the constructor where order is important. This is difficult to detect changes and preserve when a client may add/remove/switch any dependency during the upgrade process.

ie. `constructor(firstService, secondService)` IS NOT THE SAME AS `constructor(secondService, firstService)`

With the injection method, we no longer need to worry about order which significantly reduces the complexity of doing our migrations.

ie.

```
firstService = inject(FirstService)
secondService = inject(SecondService)
```

IS THE SAME AS

```
secondService = inject(SecondService)
firstService = inject(FirstService)
```

Reply:
Switched to constructor syntax. Thanks for the explanation, I appreciate it